metricbeat/helper/server/http/http.go

```diff
@@ -109,50 +91,8 @@ func (h *HttpServer) Stop() {
 	close(h.done)
 	h.stop()
 	h.server.Shutdown(h.ctx)
-	close(h.eventQueue)
 }
 
 func (h *HttpServer) GetEvents() chan server.Event {
-	return h.eventQueue
-}
-
-func (h *HttpServer) handleFunc(writer http.ResponseWriter, req *http.Request) {
-	switch req.Method {
-	case "POST":
-		meta := server.Meta{
-			"path": req.URL.String(),
-		}
-
-		contentType := req.Header.Get("Content-Type")
-		if contentType != "" {
-			meta["Content-Type"] = contentType
-		}
-
-		body, err := ioutil.ReadAll(req.Body)
-		if err != nil {
-			logp.Err("Error reading body: %v", err)
-			http.Error(writer, "Unexpected error reading request payload", http.StatusBadRequest)
-			return
-		}
-
-		payload := common.MapStr{
-			server.EventDataKey: body,
-		}
-
-		event := &HttpEvent{
-			event: payload,
-			meta:  meta,
-		}
-		h.eventQueue <- event
-		writer.WriteHeader(http.StatusAccepted)
-
-	case "GET":
-		writer.WriteHeader(http.StatusOK)
-		if req.TLS != nil {
-			writer.Write([]byte("HTTPS Server accepts data via POST"))
-		} else {
-			writer.Write([]byte("HTTP Server accepts data via POST"))
-		}
-
-	}
+	return make(chan server.Event)
```

Comment:
I guess this method can be removed now?

Reply:
will not be an issue with `NewHttpServerWithHandler`